src/styles/index.scss

```diff
@@ -1,3 +1,82 @@
+* {
+  box-sizing: border-box;
+}
+
 body {
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
   margin: 0;
+  font-family: Arial, sans-serif;
+  font-size: 30px;
+}
+
+.calendar__day {
+  background-color: #eee;
+  border: 1px solid #000;
+  justify-content: center;
+  align-items: center;
+  width: 100px;
+  height: 100px;
+  transition:
+    background-color 0.5s,
+    transform 0.5s;
+  display: flex;
+
+  &:hover {
+    cursor: pointer;
+    background-color: #ffbfcb;
+    transform: translateY(-20px);
+  }
+
+  @for $i from 1 through 31 {
+    &:nth-child(#{$i}):before {
+      content: '#{$i}';
+    }
+  }
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  width: 726px;
+  padding: 10px;
+
+  &--start-day-mon .calendar__day:first-child {
+    margin-left: 0;
+  }
+
+  &--start-day-tue .calendar__day:first-child {
+    margin-left: 101px;
+  }
+
+  &--start-day-wed .calendar__day:first-child {
+    margin-left: 202px;
+  }
+
+  &--start-day-thu .calendar__day:first-child {
+    margin-left: 303px;
+  }
+
+  &--start-day-fri .calendar__day:first-child {
+    margin-left: 404px;
+  }
+
+  &--start-day-sat .calendar__day:first-child {
+    margin-left: 505px;
+  }
+
+  &--start-day-sun .calendar__day:first-child {
+    margin-left: 606px;
+  }
+
+  &--month-length-28 .calendar__day:nth-child(n + 29),
```

Comment:
The `&--month-length-28 .calendar__day:nth-child(n + 29)` and similar selectors for other month lengths can be simplified. Since `&--month-length-28` implies that there are 28 days in the month, you can target `:nth-child(n+29)` directly without the need for `&--month-length-29`, `&--month-length-30`, and `&--month-length-31`. This would make the code cleaner and more understandable.